### PR TITLE
Delete users

### PR DESCRIPTION
**Pull request by joseivanlopez**
## Problem

YaST users client allows to delete users. But *Y2Users* classes does not support deleting users yet.

* Part of https://jira.suse.com/browse/SLE-20563
* https://trello.com/c/XcAKT8r0/2657-3-deleting-users-with-y2users

## Solution

Extend *Y2Users::Linux::Writer* in order to support deleting users.

BONUS: the writer was split into different actions. 

NOTE: this PR is part of the work accumulated in *y2users* branch.

## Testing

* Unit tests added.
* Manually tested.

### CODE REVIEW

**Review by ancorgs — APPROVED** [review]
LGTM

File: test/lib/y2users/linux/delete_user_action_test.rb [hunk 1]
```diff
@@ -63,13 +63,44 @@
...
+    context "when the command for deleting the user fails" do
+      before do
+        allow(Yast::Execute).to receive(:on_target!)
+          .and_raise(Cheetah::ExecutionFailed.new(nil, double(exitstatus: exit_status), nil, nil))
+      end
+
+      context "and error is because the user is logged in" do
+        let(:exit_status) { 8 }
+
+        it "returns a filed result with issue for logged in user" do
```
Comment (dgdavid) [inline test/lib/y2users/linux/delete_user_action_test.rb:75]: NP: failed result

File: test/lib/y2users/linux/delete_user_action_test.rb [hunk 2]
```diff
@@ -63,13 +63,44 @@
...
+
+          expect(result.success?).to eq(false)
+          expect(result.issues.first.message).to match(/is currently logged in/)
+        end
+      end
+
+      context "and error is because some files cannot be deleted" do
+        let(:exit_status) { 12 }
+
+        it "returns a filed result with issue for files that cannot deleted" do
```
Comment (dgdavid) [inline test/lib/y2users/linux/delete_user_action_test.rb:86]: NP: failed result

File: test/lib/y2users/linux/delete_user_action_test.rb [hunk 3]
```diff
@@ -63,13 +63,44 @@
...
-      expect(result.issues).to_not be_empty
+          expect(result.success?).to eq(false)
+          expect(result.issues.first.message).to match(/home or mail spool/)
+        end
+      end
+
+      context "and error is because other reason" do
+        let(:exit_status) { 1 }
+
+        it "returns a filed result with issue for deleting user" do
```
Comment (dgdavid) [inline test/lib/y2users/linux/delete_user_action_test.rb:97]: NP: failed result

**Review by dgdavid — APPROVED** [review]
LGTM, just a few typos.

**Pull request merged**